src/instruments/mongo_tracer.rs

```diff
@@ -0,0 +1,429 @@
+use std::{
+    io::Read,
+    path::{Path, PathBuf},
+    process::{Child, Command, Stdio},
+    str::FromStr,
+    thread,
+};
+
+use reqwest::Client;
+use tokio::fs;
+use url::Url;
+
+use crate::{helpers::get_env_variable, prelude::*};
+
+use super::MongoDBConfig;
+
+#[derive(Debug)]
+pub struct UserInput {
+    mongo_uri: String,
+    uri_env_name: String,
+}
+
+#[derive(Debug)]
+pub struct MongoTracer {
+    process: Option<Child>,
+    server_address: String,
+    profile_folder: PathBuf,
+    proxy_mongo_uri: String,
+    user_input: Option<UserInput>,
+}
+
+fn dump_tracer_log(mut stream: impl Read) -> Result<()> {
+    let mut buf = [0u8; 1024];
+    loop {
+        let num_read = stream.read(&mut buf)?;
```

Comment:
Add a comment mentioning that getting full lines here is a non deterministic behavior depending on the flushes of the child process.